src/util/include/util/serializing/ObjectOutputStream.h

```diff
@@ -35,18 +33,24 @@ class ObjectOutputStream {
     void writeString(const char* str);
     void writeString(const std::string& s);
 
-    void writeData(const void* data, int len, int width);
-
     template <typename T>
     void writeData(const std::vector<T>& data);
 
     /// Writes the raw image data to the output stream.
     void writeImage(const std::string_view& imgData);
 
-    GString* getStr();
+    std::string getStr();
```

Comment:
```suggestion
    std::string const& str();
```
If this is binary data:

 ```suggestion
    std::span<std::byte> data();
```
or, since we don't have c++20 yet:
```suggestion
    std::vector<std::byte> data();
```

Reply:
Change to `std::vector<std::byte> const& getData();`